### PR TITLE
Use the correct published date instead of created at date for ordering

### PR DESCRIPTION
**Pull request by ahmgeek**

### CODE REVIEW

File: backend/db/migrate/20150112104605_add_published_at_to_article.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AddPublishedAtToArticle < ActiveRecord::Migration
+  def change
+    add_column :articles, :published_at, :datetime
+
+    Article.find_each do |article|
+      article.published_at = article.created_at if article.published = true
```
Comment (mkhatib) [inline backend/db/migrate/20150112104605_add_published_at_to_article.rb]: drop the "= true"

File: backend/app/controllers/api/v1/articles_controller.rb [hunk 2]
```diff
@@ -39,6 +39,7 @@ def create
   def update
     @article = Article.find(params[:id])
     authorize @article
+    @article.published_at = Time.now if @article.published_at == nil
```
Comment (mkhatib) [inline backend/app/controllers/api/v1/articles_controller.rb]: nitpik: maybe use .nil? method instead of == nil?

**Comment by mkhatib** [comment]
Dude this looks awesome! Great work!  Had some comments

**Comment by ahmgeek** [comment]
thanks for the tips :blush: gonna clean it.

File: backend/app/models/article.rb [hunk 3]
```diff
@@ -1,9 +1,11 @@
 class Article < ActiveRecord::Base
   include Utils
 
+  before_save:published_post
```
Comment (mkhatib) [inline backend/app/models/article.rb]: Space after :

File: backend/app/models/article.rb [hunk 4]
```diff
@@ -18,12 +20,20 @@ class Article < ActiveRecord::Base
...
   scope :drafts, -> { where(published: false) }
 
+
   # Instance Methods.
   def publish!
     self.published = true
     self.save
   end
 
+  #Add published at date only for the published posts
```
Comment (mkhatib) [inline backend/app/models/article.rb]: Space after # and . at the end of the comment

Comment (ahmgeek) [inline backend/app/models/article.rb]: لامؤخذه :D

**Comment by mkhatib** [comment]
Thanks so much! Awesome work and sorry for the back and forth! :dancers:

**Comment by ahmgeek** [comment]
no no no thanks for the guidance buddy :baby_symbol: :lollipop:

**Pull request merged**